python/cudf/cudf/core/indexing.py

```diff
@@ -173,7 +173,7 @@ def _loc_to_iloc(self, arg):
 
         else:
             arg = Series(column.as_column(arg))
-            if arg.dtype in [np.bool, np.bool_]:
+            if arg.dtype in (bool, np.bool_):
```

Comment:
```suggestion
            if arg.dtype in {bool, np.bool_}:
```

Reply:
I doubt something weird is happening here, see the example below. I can do a lookup in tuple or list, but would fail in dict:

```python
>>> import numpy as np
>>> list_ = [bool, np.bool_]
>>> set_ = {bool, np.bool_}
>>> tuple_ = (bool, np.bool_)
>>> np.dtype('bool')
dtype('bool')
>>> np.dtype('bool') in list_
True
>>> np.dtype('bool') in set_
False
>>> np.dtype('bool') in tuple_
True
```

Reply:
yes, then the check would need to be changed to ```if arg.dtype.type in {bool, np.bool_}```

Reply:
Got it, didn't realize you needed non hash based comparison here. Resolving.